app/models/compute_resources/foreman/model/vmware.rb

```diff
@@ -362,16 +372,25 @@ def test_connection(options = {})
     def parse_args(args)
       args = args.deep_symbolize_keys
 
+      # set the default type first and then receive the configured type
+      scsi_controller_type = scsi_controller_default_type
+      if args[:scsi_controller_type].present?
+        scsi_controller_type = args.delete(:scsi_controller_type)
+      end
+
+      scsi_controllers = Hash.new
+      args[:volumes_attributes].each do |k,v|
```

Comment:
This gives me an `NoMethodError: undefined method `each' for nil:NilClass` when trying to create a new vm though the UI.